targon/internal/targon/callbacks.go

```diff
@@ -116,12 +116,56 @@ func AddBlockCallbacks(v *boilerplate.BaseChainSubscriber, c *Core) {
 			return
 		}
 		if c.Deps.Mongo != nil {
-			err := SyncMongo(c, int(h.Number))
+			minerInfo := MinerInfo{Core: c, Block: int(h.Number)}
+			err := SyncMongo(c, minerInfo)
```

Comment:
Sync mongo should only ever be called once per internval. Your calling it here with a partial struct, and then later with another partial struct